### DISCUSSION TITLE
Using Realtime in iOS app

### DISCUSSION THREAD

**Discussion by ThePuscher** [appwrite/appwrite#3853] [Q&A]
I want to get started using Realtime in my iOS app using the following code:

```
import Foundation
import Appwrite

class AppWrite {
    let client: Client
    let account: Account
    var session: Session?
    let realtime: Realtime
    
    init() {
        client = Client()
            .setEndpoint("http://localhost/v1")
            .setProject("<id>")
            .setSelfSigned(true) // For self signed certificates, only use for development
        account = Account(client)
        realtime = Realtime(client)
        
        let _ = realtime.subscribe(channels: ["databases.[<id>].collections.[i<id>].documents"]) { response in
                print(String(describing: response))
        }
    }

    ...

}
```

The following error occurs:
`.../SourcePackages/checkouts/sdk-for-apple/Sources/Appwrite/WebSockets/WebSocketClient.swift:233: warning run: Thread running at QOS_CLASS_USER_INTERACTIVE waiting on a lower QoS thread running at QOS_CLASS_DEFAULT. Investigate ways to avoid priority inversions`

Am I using the subscribe function in a wrong way?

**Comment by stnguyen90** [comment]
@ThePuscher would you please try taking a look at [this](https://github.com/appwrite/playground-for-apple-swiftui/blob/944d12aea904452ae42b66c8e074b4ac047539ff/PlaygroundForApple/PlaygroundViewModel.swift#L140) sample to see if it helps?
